src/App.tsx

```diff
@@ -1,223 +1,139 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/Filter';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { ErrorMessage } from './types/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = React.useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [fileterType, setFilterType] = React.useState(FilterType.All);
+  const [title, setTitle] = React.useState('');
+  const [selectedId, setSelectedId] = React.useState<number[]>([]);
+  const [isAdding, setisAdding] = React.useState(false);
+  const user = React.useContext(AuthContext);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    getTodosFromServer(user.id);
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (fileterType) {
+        case FilterType.Active:
+          return !todo.completed;
+        case FilterType.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [todos, fileterType]);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+    if (!title || !user) {
+      setErrorMessage(ErrorMessage.ErrorTitle);
+
+      return;
+    }
+
+    setisAdding(true);
+
+    try {
+      const postTodo = await addTodo(title, user.id);
+
+      setTodos([...todos, postTodo]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos